cpu/samd21/periph/pwm.c

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (C) 2014 Hamburg University of Applied Sciences
+ *               2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_samd21
+ * @{
+ *
+ * @file
+ * @brief       Low-level PWM driver implementation
+ *
+ * @author      Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+#include <string.h>
+
+#include "log.h"
+#include "cpu.h"
+#include "board.h"
+#include "periph/pwm.h"
+#include "periph_conf.h"
+
+/* ignore file in case no PWM devices are defined */
+#if PWM_NUMOF
+
+static inline int _num(pwm_t dev)
+{
+    return ((int)(pwm_config[dev].dev) & 0xc00) >> 10;
```

Comment:
simple: if you compare the bases address of all available TCC devices and look for their differences, youll see that they differ only in bits 10 and 11. So masking these bits and shifting them to the LSB positions, youll have upcounting integers 0, 1, 2 for the three TCC devices.